tests/protocol/mockwiredata.py

```diff
@@ -289,44 +208,125 @@ def replace_xml_attribute_value(xml_document, element_name, attribute_name, attr
             raise Exception("Could not match attribute '{0}' of element '{1}'".format(attribute_name, element_name))
         return new_xml_document
 
+    @staticmethod
+    def replace_xml_attribute_value_on_matching_element(xml_document, element_name, identifying_attr_str, attribute_name, attribute_value):
+
+        # Two cases: the attribute we're using to identify the element might be before the attribute we're replacing,
+        # or it might be after. We need to handle both.
```

Comment:
Could you add a quick example in the comment section of this method? I'm not sure what you're referring to just based on reading this sentence.

Reply:
I've pushed a change; let me know if the examples help.

Reply:
Gotcha! I actually didn't know the sequence of the attributes was arbitrary, but it's something we can't control. Nice catch!